File: MyFullProject/spring-backend/springbackend/src/main/java/com/example/springbackend/dto/ChatHeaderResponse.java
package com.example.springbackend.dto;

import java.util.List;
import lombok.Getter;
import lombok.Setter;
@Getter
@Setter

public class ChatHeaderResponse {

    private boolean privateChat;

    private Long activityId;

    private String activityTitle;

    private Integer participantCount;

    private boolean isGroup;

    private Long otherUserId;

    private String otherUserName;

    private Long otherUserProfileImageId;

    private List<ParticipantDto> participants;

 
    // getters/setters


    public boolean isGroup() {
    return isGroup;
    }

    public void setIsGroup(boolean isGroup) {
        this.isGroup = isGroup;
    }
}

File: MyFullProject/android-app/MyApplication/app/src/main/java/com/example/myapplication/models/ChatHeaderResponse.kt
package com.example.myapplication.models

data class ChatHeaderResponse(

    val activityTitle: String?,

    val participantCount: Int,

    val participants: List<Participant>,

    val isGroup: Boolean,

    val privateChat: Boolean,

    val otherUserId: Long?,

    val otherUserName: String?,

    val otherUserProfileImageId: Long?

)